### ISSUE TITLE
sizehint causing the app to block?

### ISSUE THREAD

**Issue by jdonnerstag** [andyferris/Dictionaries.jl#37]
Hi, 

I extracted the below snippet into a separate file, and insert it from REPL to validate/demonstrate the issue. I'm on a Win10 laptop running Julia 1.5.3, the issue is in my env 100% repeatable.

```julia
    using Dictionaries

    function unique_index_from_in_memory(len::Int)
        sizehint = floor(Int, len * 1.2)
        d = Dictionary{Int,Int}(; sizehint=sizehint)

        count = 1
        @show sizehint, len
        for value in 1:len # values(this, field)
            println(Base.stderr, "$count: $value")
            #insert!(d, value, count)
            set!(d, value, count)
            count += 1
        end

        return d
    end

    function test_fn()
        unique_index_from_in_memory(76525)
    end

    test_fn()
```

What is the issue: The app blocks/stops progressing forever at a certain point. 
My output is as follows. I stopped the app after 5 mins, by hitting ctrl-c several times (hitting it ones is not enough).  I don't get any stacktrace.
The app always stops at the same position: 153
It depends on the value of sizehint, where it stops.
Without any sizehint, it runs without issues.
The issue is independent from set!() and insert!()
```
     148: 148
     149: 149
     150: 150
     151: 151
     152: 152
     153: 153
     WARNING: Force throwing a SIGINT
```

Any idea why that might happen?

UPDATE: properly formatted the code

**Comment by andyferris** [comment]
Thanks I’ll have to have a closer look. The sizehint should control the initial size of the backing arrays, and not much else. Unless you are running out of RAM I’m not sure what the issue might be.

**Comment by andyferris** [comment]
OK I found a very silly bug and have fixed in #38.

**Comment by andyferris** [comment]
@jdonnerstag if you update your packages this issue should be fixed.

Thanks for the report and let us know if you have any other problems.
